feature: heterogeneous lookup for ordered containers

impugachev opened the pull request:
1. Deleted useless `is_transparent` from `map`.
2. Fixed type safety of `map`.
3. Added heterogenous lookup for `set`.

`CompareKey` isn't overcomplication, as I said before. It's an adaptor for bound functions, so that there is no need to write additional overloads.

impugachev commented:
Also, can we release a new version after this PR?

serge-sans-paille commented on tests/test_set.cpp:
```diff
@@ -279,6 +279,22 @@ TEST_CASE("frozen::set of frozen::set", "[set]") {
+constexpr inline bool operator<(int a, Foo const &b) { return a < b.x; }
+
+TEST_CASE("frozen::set heterogeneous container", "[set]") {
+  constexpr auto set = frozen::make_set<Foo, std::less<>>({Foo{1}, Foo{2}, Foo{3}});
```
could you also add a test that intializes from a plain C array? (for instance with the same content as `set` then we can add a check that tests for equality).

impugachev replied:
Did I understand correctly that you mean a test that will check that `make_set` makes the same objects from a C array and `std::array`? Or just the same test as this, only with initialization from C array?

serge-sans-paille commented:
> Did I understand correctly that you mean a test that will check that make_set
> makes the same objects from a C array and std::array? Or just the same test as
> this, only with initialization from C array?

both :-)

impugachev commented:
OK, update:
1. Deleted `type_traits` with unused `void_t` and `is_transparent`.
2. Made tests for `make_set` more explicit.
3. Added second set to transparency test.

serge-sans-paille commented:
Thanks!

serge-sans-paille commented:
(release done, tagged as 1.1.0)
